src/rv32_template.c

```diff
@@ -4,6 +4,8 @@
  */
 
 /* Internal */
+#include <stdio.h>
+#include "riscv_private.h"
```

Comment:
As the header name suggests, this template is intended for both interpreter and JIT compiler. Whenever feasible, we aim to eliminate the need for including this header. Additionally, each function's body should be concise, making use of macros and helper functions from `src/emulate.c`. This approach will help minimize the proposed changes in the `src/rv32_template.c` file.